common/persistence/sql/sqlExecutionManager.go

```diff
@@ -476,8 +479,9 @@ LIMIT ?`
 		`
 FROM timer_tasks WHERE
 shard_id = ? AND
-visibility_timestamp >= ? AND
-visibility_timestamp < ?`
+((visibility_timestamp >= ? AND task_id >= ?) OR visibility_timestamp > ?) AND
+visibility_timestamp < ?
```

Comment:
This condition looks different from the one i'm using. Which would be:
visibility_timestamp < ? and visibility_timestamp >=? and (visibility_timestamp > ? OR task_id >= ?)

Reply:
its different because - timer tasks are stored / queried in natural order (asc) whereas with your other case - data is queried in reverse sorted order